CI/macos/appcast_download.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+import requests
+import xmltodict
+
+
+def download_build(url):
+    print('Downloading', url)
+    filename = url.rpartition('/')[2]
+    r = requests.get(url)
+    if r.status_code == 200:
+        with open(f'artifacts/{filename}', 'wb') as f:
+            f.write(r.content)
+
+
+def read_appcast(url, fallback):
+    r = requests.get(url)
+    if r.status_code == 404:
+        r = requests.get(fallback)
+        if r.status_code == 404:
+            return
+
+    filename = url.rpartition('/')[2]
+    with open(f'builds/{filename}', 'wb') as f:
+        f.write(r.content)
+
+    appcast = xmltodict.parse(r.content, force_list=('item',))
```

Comment:
Code continues unchecked even if target file wasn't written, resulting in undefined state.

Reply:
I would expect an exception to be thrown if writing the file fails for whatever reason, which doesn't have a pretty error here, but would abort the script execution and set a non-zero exit code.